test/tutorial/cpp/asyncfinish/Makefile

```diff
@@ -0,0 +1,17 @@
+include $(HCLIB_ROOT)/include/hclib.mak
```

Comment:
Lots of questions on this makefile (and the others that mirror it):

1. Why does clean-ob/clean remove .dSYM? We shouldn't be creating any libraries, right? Or is this some mac-specific file I'm not familiar with?
2. Same question, but for *.o. Doesn't seem like we're creating object files.
3. Why do we need both clean and clean-obj? Seems like just having clean is good enough.
4. Why does 'all' trigger 'clean' and 'clean-obj'? I'd personally prefer just 'all: $(EXE)'